lib/ansible/modules/cloud/amazon/elasticache_parameter_group.py

```diff
@@ -0,0 +1,336 @@
+    - name: 'Create a test parameter group'
+      elasticache_parameter_group:
+        name: 'test-param-group'
+        group_family: 'redis3.2'
+        description: 'This is a cache parameter group'
+        state: 'present'
+    - name: 'Modify a test parameter group'
+      elasticache_parameter_group:
+        name: 'test-param-group'
+        values:
+          - ['activerehashing', 'yes']
+          - ['client-output-buffer-limit-normal-hard-limit', 4]
+        state: 'present'
+    - name: 'Reset all modifiable parameters for the test parameter group'
+      elasticache_parameter_group:
+        name: 'test-param-group'
+        state: reset
+    - name: 'Delete a test parameter group'
+      elasticache_parameter_group:
+        name: 'test-param-group'
+        state: 'absent'
+"""
+
+RETURN = """
+elasticache:
+    description: cache parameter group information and response metadata
+    returned: always
+    type: dict
+    sample:
+        "cache_parameter_group": {
+            "cache_parameter_group_family": "redis3.2",
+            "cache_parameter_group_name": "test-please-delete",
+            "description": "initial description"
+        },
+        "response_metadata": {
+            "http_headers": {
+                "content-length": "562",
+                "content-type": "text/xml",
+                "date": "Mon, 06 Feb 2017 22:14:08 GMT",
+                "x-amzn-requestid": "947291f9-ecb9-11e6-85bd-3baa4eca2cc1"
+            },
+            "http_status_code": 200,
+            "request_id": "947291f9-ecb9-11e6-85bd-3baa4eca2cc1",
+            "retry_attempts": 0
+        }
+changed:
+    description: if the cache parameter group has changed
+    returned: always
+    type: bool
+    sample:
+        "changed": true
+"""
+
+
+from ansible.module_utils.ec2 import boto3_conn
+
+try:
+    import boto3
+    import botocore
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+def create(module, conn, name, group_family, description):
+    """ Create ElastiCache parameter group. """
+    try:
+        response = conn.create_cache_parameter_group(CacheParameterGroupName=name, CacheParameterGroupFamily=group_family, Description=description)
+        changed = True
+    except boto.exception.ClientError as e:
+        module.fail_json(msg=e.message)
+    return response, changed
+
+def delete(module, conn, name):
+    """ Delete ElastiCache parameter group. """
+    try:
+        conn.delete_cache_parameter_group(CacheParameterGroupName=name)
+        response = {}
+        changed = True
+    except boto.exception.ClientError as e:
+        module.fail_json(msg=e.message)
+    return response, changed
+
+def make_current_modifiable_param_dict(conn, name):
+    """ Gets the current state of the cache parameter group and creates a dict with the format: {ParameterName: [Allowed_Values, DataType, ParameterValue]}"""
+    current_info = get_info(conn, name)
+    if current_info is False:
+        return False
```

Comment:
Instead of returning False, I think this function should raise an exception if the parameter group can't be found. This function only gets run to get the current parameters from an existing group, so failing silently like this will lead to problems later in the script. This value gets passed to [check_valid_modification](https://github.com/ansible/ansible/pull/21023/files#diff-2faf68f29dd9e539bec6093d1ac544e3R170) and used as a dict without a type check. It makes sense to me to have this raise an exception & fail the module if the param group can't be found.